flight/uavobjectlib/Makefile

```diff
@@ -66,6 +66,9 @@ LDFLAGS += -Wl,--fatal-warnings
 SRC += $(filter-out $(OPUAVSYNTHDIR)/uavobjectsinit.c,$(wildcard $(OPUAVSYNTHDIR)/*.c))
 NO_AUTO_UAVO = YES
 
+# Override the local version of RTOS before entering firmware-common.mk
+override RTOS = UNDEFINED_FOR_UAVOBJECT_LIB
```

Comment:
Still not terribly happy with this approach, but it's cleaner now thanks to @mlyle's input. Unless someone disagrees, I'm good to leave it as it is for a while until we have other targets that also need to unset this variable.